Controller/Payment/CancelOrder.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+namespace Alma\MonthlyPayments\Controller\Payment;
+
+use Alma\API\Entities\Order;
+use Alma\API\RequestError;
+use Alma\MonthlyPayments\Helpers\Logger;
+use InvalidArgumentException;
+use Magento\Framework\App\Action\Action;
+use Magento\Framework\App\Action\Context;
+use Magento\Framework\App\ResponseInterface;
+use Alma\MonthlyPayments\Helpers\PaymentValidation;
+use Magento\Sales\Api\Data\OrderInterface;
+use Magento\Sales\Api\OrderManagementInterface;
+use Magento\Sales\Model\OrderFactory;
+
+class CancelOrder extends Action
+{
+    /**
+     * @var Logger
+     */
+    private $logger;
+    /**
+     * @var PaymentValidation
+     */
+    private $paymentValidation;
+    /**
+     * @var \Alma\API\Entities\Payment
+     */
+    private $almaPayment;
+    /**
+     * @var OrderFactory
+     */
+    private $orderFactory;
+    /**
+     * @var OrderManagementInterface
+     */
+    private $orderManagement;
+
+    public function __construct(
+        Context $context,
+        Logger $logger,
+        PaymentValidation $paymentValidation,
+        OrderFactory $orderFactory,
+        OrderManagementInterface $orderManagement
+    )
+    {
+        parent::__construct($context);
+
+        $this->logger = $logger;
+        $this->paymentValidation = $paymentValidation;
+        $this->orderFactory = $orderFactory;
+        $this->orderManagement = $orderManagement;
+    }
+
+    public function execute()
+    {
+        $this->logger->info('In Cancel',[$this->getRequest()->getParams()]);
```

Comment:
I'am not sure about this log necessity (according to following log info you had removed in `Helpers/Eligibliy` for example)

Reply:
Remove no error log